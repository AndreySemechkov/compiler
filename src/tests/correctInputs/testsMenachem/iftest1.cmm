main
var a0,a1:real;
    b0,b1:integer;
{
	a0 = 1.5;
	a1 = 2.5;
	b0 = 1;
	b1 = 2;
}
//	if b0 < b1 then {
//		write(b0);
//		write("<");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if b0 > b1 then {
//		write(b0);
//		write(">");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if b0 <> b1 then {
//		write(b0);
//		write("<>");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if b0 == b1 then {
//		write(b0);
//		write("==");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if b0 >= b1 then {
//		write(b0);
//		write(">=");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if b0 <= b1 then {
//		write(b0);
//		write("<=");
//		write(" ");
//		write(b1);
//		write("\n");
//	}
//	if a0 < a1 then {
//		write(a0);
//		write("<");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if a0 > a1 then {
//		write(a0);
//		write(">");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if a0 <> a1 then {
//		write(a0);
//		write("<>");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if a0 == a1 then {
//		write(a0);
//		write("==");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if a0 >= a1 then {
//		write(a0);
//		write(">=");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if a0 <= a1 then {
//		write(a0);
//		write("<=");
//		write(" ");
//		write(a1);
//		write("\n");
//	}
//	if (b0 < b1 && a0 < a1) then
//	{
//		write(b0);
//		write(" < ");
//		write(b1);
//		write(" and ");
//		write(a0);
//		write(" < ");
//		write(a1);
//		write("\n");
//	}
//	else if (b0 >= b1 && a0 < a1) then
//	{
//		write(b0);
//		write(" >= ");
//		write(b1);
//		write(" and ");
//		write(a0);
//		write(" < ");
//		write(a1);
//		write("\n");
//	}
//	else if (b0 < b1 && a0 >= a1) then
//	{
//		write(b0);
//		write(" < ");
//		write(b1);
//		write(" and ");
//		write(a0);
//		write(" >= ");
//		write(a1);
//		write("\n");
//	}
//	else {
//		write(b0);
//		write(" >= ");
//		write(b1);
//		write(" and ");
//		write(a0);
//		write(" >= ");
//		write(a1);
//		write("\n");
//	}
//	if (b0 < b1 || a0 < a1) then
//	{
//		write(b0);
//		write(" < ");
//		write(b1);
//		write(" or ");
//		write(a0);
//		write(" < ");
//		write(a1);
//		write("\n");
//	}
//	else 
//	{
//		write(b0);
//		write(" >= ");
//		write(b1);
//		write(" and ");
//		write(a0);
//		write(" >= ");
//		write(a1);
//		write("\n");
//	}
//}
