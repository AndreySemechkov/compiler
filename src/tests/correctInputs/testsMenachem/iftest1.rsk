<header>
<main>
<unimplemented>
<implemented>
</header>
UJUMP 2
COPYI I0 0
COPYI I1 0
COPYI I2 0
COPYI I3 0
COPYI I4 0
ADD2I I2 I2 2
ADD2I I4 I4 2
COPYR R0 1.5
STORR R0 0 I3
COPYR R1 2.5
STORR R1 1 I3
COPYI I5 1
STORI I5 0 I1
COPYI I6 2
STORI I6 1 I1
LOADI I7 I1 0
LOADI I8 I1 1
SLETI I9 I7 I8
BREQZ I9 33
UJUMP 22
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I10 I1 0
PRNTI I10
PRNTC 60
PRNTC 32
LOADI I11 I1 1
PRNTI I11
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SGRTI I7 I5 I6
BREQZ I7 49
UJUMP 38
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I8 I1 0
PRNTI I8
PRNTC 62
PRNTC 32
LOADI I9 I1 1
PRNTI I9
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SNEQI I7 I5 I6
BREQZ I7 66
UJUMP 54
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I8 I1 0
PRNTI I8
PRNTC 60
PRNTC 62
PRNTC 32
LOADI I9 I1 1
PRNTI I9
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SEQUI I7 I5 I6
BREQZ I7 83
UJUMP 71
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I8 I1 0
PRNTI I8
PRNTC 61
PRNTC 61
PRNTC 32
LOADI I9 I1 1
PRNTI I9
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SLETI I7 I5 I6
BNEQZ I7 100
UJUMP 88
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I8 I1 0
PRNTI I8
PRNTC 62
PRNTC 61
PRNTC 32
LOADI I9 I1 1
PRNTI I9
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SGRTI I7 I5 I6
BNEQZ I7 117
UJUMP 105
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I8 I1 0
PRNTI I8
PRNTC 60
PRNTC 61
PRNTC 32
LOADI I9 I1 1
PRNTI I9
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I5 R0 R1
BREQZ I5 133
UJUMP 122
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 60
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SGRTR I5 R0 R1
BREQZ I5 149
UJUMP 138
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 62
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SNEQR I5 R0 R1
BREQZ I5 166
UJUMP 154
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 60
PRNTC 62
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SEQUR I5 R0 R1
BREQZ I5 183
UJUMP 171
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 61
PRNTC 61
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I5 R0 R1
BNEQZ I5 200
UJUMP 188
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 62
PRNTC 61
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADR R0 I3 0
LOADR R1 I3 1
SGRTR I5 R0 R1
BNEQZ I5 217
UJUMP 205
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADR R2 I3 0
PRNTR R2
PRNTC 60
PRNTC 61
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SLETI I7 I5 I6
BREQZ I7 252
UJUMP 222
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I8 R0 R1
BREQZ I8 252
UJUMP 227
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I9 I1 0
PRNTI I9
PRNTC 32
PRNTC 60
PRNTC 32
LOADI I10 I1 1
PRNTI I10
PRNTC 32
PRNTC 97
PRNTC 110
PRNTC 100
PRNTC 32
LOADR R2 I3 0
PRNTR R2
PRNTC 32
PRNTC 60
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
UJUMP 350
LOADI I5 I1 0
LOADI I6 I1 1
SLETI I7 I5 I6
BNEQZ I7 288
UJUMP 257
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I8 R0 R1
BREQZ I8 288
UJUMP 262
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I9 I1 0
PRNTI I9
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADI I10 I1 1
PRNTI I10
PRNTC 32
PRNTC 97
PRNTC 110
PRNTC 100
PRNTC 32
LOADR R2 I3 0
PRNTR R2
PRNTC 32
PRNTC 60
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
UJUMP 350
LOADI I5 I1 0
LOADI I6 I1 1
SLETI I7 I5 I6
BREQZ I7 324
UJUMP 293
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I8 R0 R1
BNEQZ I8 324
UJUMP 298
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I9 I1 0
PRNTI I9
PRNTC 32
PRNTC 60
PRNTC 32
LOADI I10 I1 1
PRNTI I10
PRNTC 32
PRNTC 97
PRNTC 110
PRNTC 100
PRNTC 32
LOADR R2 I3 0
PRNTR R2
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
UJUMP 350
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I5 I1 0
PRNTI I5
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADI I6 I1 1
PRNTI I6
PRNTC 32
PRNTC 97
PRNTC 110
PRNTC 100
PRNTC 32
LOADR R0 I3 0
PRNTR R0
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADR R1 I3 1
PRNTR R1
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
LOADI I5 I1 0
LOADI I6 I1 1
SLETI I7 I5 I6
BREQZ I7 355
UJUMP 360
LOADR R0 I3 0
LOADR R1 I3 1
SLETR I8 R0 R1
BREQZ I8 384
UJUMP 360
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I9 I1 0
PRNTI I9
PRNTC 32
PRNTC 60
PRNTC 32
LOADI I10 I1 1
PRNTI I10
PRNTC 32
PRNTC 111
PRNTC 114
PRNTC 32
LOADR R2 I3 0
PRNTR R2
PRNTC 32
PRNTC 60
PRNTC 32
LOADR R3 I3 1
PRNTR R3
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
UJUMP 410
ADD2I I2 I2 0
ADD2I I4 I4 0
LOADI I5 I1 0
PRNTI I5
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADI I6 I1 1
PRNTI I6
PRNTC 32
PRNTC 97
PRNTC 110
PRNTC 100
PRNTC 32
LOADR R0 I3 0
PRNTR R0
PRNTC 32
PRNTC 62
PRNTC 61
PRNTC 32
LOADR R1 I3 1
PRNTR R1
PRNTC 10
SUBTI I2 I2 0
SUBTI I4 I4 0
SUBTI I2 I2 2
SUBTI I4 I4 2
HALT
