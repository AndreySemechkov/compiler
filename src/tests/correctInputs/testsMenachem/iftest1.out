1 < 2 
1 <> 2 
1 <= 2 
1.500000 < 2.500000 
1.500000 <> 2.500000 
1.500000 <= 2.500000 
1  < 2  and 1.500000  < 2.500000 
1  < 2  or 1.500000  < 2.500000 
Reached Halt.
